core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -361,33 +361,97 @@ object ShufflingNonInteractiveFinalizer
   }
 }
 
-case class SubtractFeeFromOutputsFinalizer(
-    inputInfos: Vector[InputInfo],
-    feeRate: FeeUnit)
+/** Adds a future fee amount to the output with the given
+  * ScriptPubKey and subtracts that amount in equal portions
+  * from the specified change ScriptPubKeys.
+  */
+case class AddFutureFeeFinalizer(
+    spk: ScriptPubKey,
+    futureFee: CurrencyUnit,
+    changeSPKs: Vector[ScriptPubKey])
     extends RawTxFinalizer {
 
   override def buildTx(txBuilderResult: RawTxBuilderResult)(implicit
       ec: ExecutionContext): Future[Transaction] = {
-    val RawTxBuilderResult(version, inputs, outputs, lockTime) = txBuilderResult
+    val changeOutputs = txBuilderResult.outputs.filter(output =>
+      changeSPKs.contains(output.scriptPubKey))
+
+    val outputT = txBuilderResult.outputs.zipWithIndex
+      .find(_._1.scriptPubKey == spk) match {
+      case Some((output, index)) =>
+        val newOutput = output.copy(value = output.value + futureFee)
+        Success((newOutput, index))
+      case None =>
+        Failure(new RuntimeException(
+          s"Did not find expected SPK $spk in ${txBuilderResult.outputs.map(_.scriptPubKey)}"))
+    }
 
-    val witnesses = inputInfos.map(InputInfo.getScriptWitness)
-    val txWithPossibleWitness = TransactionWitness.fromWitOpt(witnesses) match {
-      case _: EmptyWitness =>
-        BaseTransaction(version, inputs, outputs, lockTime)
-      case wit: TransactionWitness =>
-        WitnessTransaction(version, inputs, outputs, lockTime, wit)
+    val outputsT = outputT.map {
+      case (newOutput, index) =>
+        val subFromChange =
+          Satoshis(futureFee.satoshis.toLong / changeOutputs.length)
+        val outputsMinusChange = txBuilderResult.outputs.map { output =>
+          if (changeSPKs.contains(output.scriptPubKey)) {
+            output.copy(value = output.value - subFromChange)
+          } else {
+            output
+          }
+        }
+
+        outputsMinusChange.updated(index, newOutput)
+    }
+
+    val txT = outputsT.map { outputs =>
+      txBuilderResult.toBaseTransaction.copy(outputs = outputs)
+    }
+
+    Future.fromTry(txT)
+  }
+}
+
+case class SubtractFromOutputFinalizer(spk: ScriptPubKey, subAmt: CurrencyUnit)
```

Comment:
nit: Scaladoc 

I'm assuming this just subtract the given amount from the output that corresponds to the given `spk` ? It might also be useful to add what contexts that is finalizer is useful for.

Reply:
I don't think this has any unit test cases